### PR TITLE
Refactor/even more coords stuff

### PR DESCRIPTION
**Pull request by Gymnasiast**
Split off from https://github.com/OpenRCT2/OpenRCT2/pull/10922

### CODE REVIEW

File: src/openrct2/peep/GuestPathfinding.cpp [hunk 1]
```diff
@@ -1624,15 +1620,11 @@ static int32_t guest_path_find_leaving_park(Peep* peep, uint8_t edges)
...
-    int16_t y = peepSpawn->y & 0xFFE0;
-    uint8_t z = peepSpawn->z / 8;
-    Direction direction = peepSpawn->direction;
+    const auto peepSpawnLoc = gPeepSpawns[chosenSpawn].ToTileStart();
+    Direction direction = peepSpawnLoc.direction;
 
-    gPeepPathFindGoalPosition = { x / 32, y / 32, z };
-    if (x == peep->NextLoc.x && y == peep->NextLoc.y)
+    gPeepPathFindGoalPosition = TileCoordsXYZ(peepSpawnLoc);
+    if (peepSpawnLoc.x == peep->NextLoc.x && peepSpawnLoc.y == peep->NextLoc.y)
```
Comment (tupaschoal) [inline src/openrct2/peep/GuestPathfinding.cpp]: Are these different objects or could the equality be simplified?

File: src/openrct2/actions/FootpathPlaceAction.hpp [hunk 2]
```diff
@@ -76,12 +76,12 @@ DEFINE_GAME_ACTION(FootpathPlaceAction, GAME_COMMAND_PLACE_PATH, GameActionResul
             return MakeResult(GA_ERROR::DISALLOWED, STR_CANT_BUILD_FOOTPATH_HERE, STR_LAND_SLOPE_UNSUITABLE);
         }
 
-        if (_loc.z / 8 < FootpathMinHeight)
+        if (_loc.z < FootpathMinHeight)
```
Comment (tupaschoal) [inline src/openrct2/actions/FootpathPlaceAction.hpp]: Isn't this a potentially different result?

Comment (Gymnasiast) [reply src/openrct2/actions/FootpathPlaceAction.hpp]: Doubt it, as it won't let you build at z coordinates that aren't multiples of 1.5 metres anyway.

File: src/openrct2/peep/GuestPathfinding.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/peep/GuestPathfinding.cpp]: `peepSpawnLoc` is a CoordsXYZD and `peep->NextLoc` is a CoordsXYZ. Comparing them directly will also compare the Z coordinate, which is not supposed to happen.

Comment (tupaschoal) [reply src/openrct2/peep/GuestPathfinding.cpp]: Ok, couldn't easily check on phone, thanks

File: src/openrct2/actions/FootpathPlaceAction.hpp (hunk 2 above)
Comment (tupaschoal) [reply src/openrct2/actions/FootpathPlaceAction.hpp]: Alright

File: src/openrct2/paint/tile_element/Paint.Surface.cpp [hunk 3]
```diff
@@ -518,7 +518,7 @@ static bool tile_is_inside_clip_view(const tile_descriptor& tile)
     if (tile.tile_element == nullptr)
         return false;
 
-    if (tile.tile_element->base_height > gClipHeight)
+    if (tile.tile_element->GetBaseZ() > gClipHeight / COORDS_Z_STEP)
```
Comment (duncanspumpkin) [inline src/openrct2/paint/tile_element/Paint.Surface.cpp]: Assume this has been picked up. Incorrect comparison.

File: src/openrct2/paint/tile_element/Paint.TileElement.cpp [hunk 4]
```diff
@@ -245,7 +245,7 @@ static void sub_68B3FB(paint_session* session, int32_t x, int32_t y)
     do
     {
         // Only paint tile_elements below the clip height.
-        if ((session->ViewFlags & VIEWPORT_FLAG_CLIP_VIEW) && (tile_element->base_height > gClipHeight))
+        if ((session->ViewFlags & VIEWPORT_FLAG_CLIP_VIEW) && (tile_element->GetBaseZ() > gClipHeight / COORDS_Z_STEP))
```
Comment (duncanspumpkin) [inline src/openrct2/paint/tile_element/Paint.TileElement.cpp]: Incorrect comparison.

**Pull request merged**